pando/utils.py

```diff
@@ -17,6 +17,13 @@ def maybe_encode(s, codec='ascii'):
     return s.encode(codec) if isinstance(s, text_type) else s
 
 
+def try_encode(s, codec='ascii'):
+    try:
+        return s.encode(codec) if isinstance(s, text_type) else s
```

Comment:
Sorry, sometimes I forget to tell myself not to do micro-optimizations (here saving a function call by inlining the code).